unity-environment/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -417,17 +417,52 @@ public virtual void CollectObservations()
     }
 
     /// <summary>
-    /// Adds a vector observation. 
-    /// Note that the number of vector observation to add
+    /// Appends float values to the vector observation.
+    /// Note that the total number of vector observation added
     /// must be the same at each CollectObservations call.
     /// </summary>
-    /// <param name="observation">The float value to add to 
+    /// <param name="observation">The value to add to 
     /// the vector observation.</param>
     internal void AddVectorObs(float observation)
     {
         _info.vectorObservation.Add(observation);
     }
+    internal void AddVectorObs(int observation)
+    {
+        _info.vectorObservation.Add((float)observation);
+    }
+    internal void AddVectorObs(Vector3 observation)
+    {
+        _info.vectorObservation.Add(observation.x);
+        _info.vectorObservation.Add(observation.y);
+        _info.vectorObservation.Add(observation.z);
+    }
+    internal void AddVectorObs(Vector2 observation)
+    {
+        _info.vectorObservation.Add(observation.x);
+        _info.vectorObservation.Add(observation.y);
+    }
+    internal void AddVectorObs(float[] observation)
+    {
+        foreach (float obs in observation)
```

Comment:
We can add a range to the list, thus not needing the foreach. Same for the list + list version below.